frontend/src/auth/__tests__/handleNotAuthorizedErrors.test.ts

```diff
@@ -39,9 +39,15 @@ describe('given the application config', () => {
       it('should redirect to the authorization server', () => {
         handleNotAuthorizedErrors(mockAppConfig)(mockRequestPromise).catch(() => {
             expect(global.window.location.replace).toHaveBeenCalledTimes(1)
-            expect(global.window.location.replace).toHaveBeenCalledWith('some-url?response_type=code&client_id=some-id&scope=some-list&redirect_uri=some-uri')
-          })
+            expect(global.window.location.replace).toHaveBeenCalledWith(
+                expect.stringContaining('some-url?response_type=code&client_id=some-id&scope=some-list&redirect_uri=some-uri')
+            )
+            // 'as any' to avoid breaking npx tsc -p tsconfig.json
+            const urlString = `'${(global.window.location.replace as any).mock.calls[0][0]}`;
+            const queryParams = urlString.split('?')[1].split('&').map(keyvalue => keyvalue.split('='))
+            expect(queryParams.filter(([key, value]) => key === 'state').length).toBeGreaterThan(0)
```

Comment:
suggestion: I think it would be a bit clearer and less error prone if we did this as follows:

```ts
const searchParams = new URL(urlString).searchParams
expect(searchParams.get('state')).toBeTruthy()
```

You'd simply need to set the mocked path to be a reasonable path, such as `http://somepath.com`

Reply:
Unfortunately new URL(urlString) throws an exception with the "fake" url, I couldn't get it to work
so I had to do it manually

Reply:
I couldn't get it to work even adding protocol and trying like "http://some.url"
new URL() throws an exception